include/highfive/bits/H5Easy_vector.hpp

```diff
@@ -0,0 +1,105 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_VECTOR_HPP
+#define H5EASY_BITS_VECTOR_HPP
+
+#include "../H5Easy.hpp"
+#include "H5Easy_misc.hpp"
+
+namespace HighFive
+{
+    namespace detail
+    {
+        namespace vector
+        {
+            template <class T, class E = void>
+            struct dump_impl
+            {
+                template <class C>
+                static HighFive::DataSet run(HighFive::File& file, const std::string& path, const std::vector<C>& data)
```

Comment:
I find the implementation quite complicated. With such level of abstraction I would expect more code reuse. Eventually there is not the need for so many templates since it seems to me that T and C are related. Could you maybe explain better this design?
@tristan0x do you see simplifications that could be done here?

Reply:
Maybe it was not the most logical choice, but I kept the structure the same everywhere. The structure is useful for Eigen/xtensor. Got rid to the `C` where I could.